### ISSUE TITLE
Experiments on one-cycle policy (super convergence)

### ISSUE THREAD

**Issue by lucaventurini** [flairNLP/flair#1282]
Hi! As in title, I'd really like to experiment with one cycle policy for the LR scheduler.

Did you already try that? I saw #286 but it seems abandoned.

**Comment by lucaventurini** [comment]
Hi,
I have implemented this policy on a local repo (it was quite easy afterall, the code of the trainer is very clean and easy to modify, kudos for that :+1: ). I used the OneCycleLR from pytorch.

While the optimizer converges quite fast to a very decent value for the loss, I still haven't seen "superconvergence" I think (since I'm still slightly above the loss of SGD trained for 500 epochs with LR decay, the default one for flair). But I have seen an unexpected behavior.

The original paper by Smith (Super-Convergence: Very Fast Training of Neural Networks Using Large Learning Rates) says that the learning rate LR and the batch size BS should scale at the same rate, and that we should pick the highest LR that does not diverge and the highest BS that fits memory. Now, I thought that BS in flair would be equal to mini_batch_size, and that doubling mini_batch_size I could double the max (non-divergent, found with the LR finder and slightly finetuned-reduced) LR and still not diverging. What I saw in my experiments, instead, it is that it's mini_batch_chunk_size that scales linearly with the LR. So in practice I'm still very limited by memory, since even with storage in cpu I cannot go beyond 8 for mini_batch_chunk_size (and so probably I will never reach the high LR suggested by Smith).

@alanakbik , any thoughts on this?

I think that if this behavior is confirmed, it affects also batch growth annealing (#1138 ), since the linear dependency between LR and BS is also behind that.

### How to reproduce:
I think this can be seen also without the One Cycle policy implemented, in theory.

1. Set up a text classification
2. Find the max LR with find_learning_rate() and fixed BS
3. Train a classifier with the max LR found; if train loss diverges, reduce LR slightly;
4. Re-train doubling LR and mini_batch_size (it should not diverge in theory, but it'll probably do)
